### ISSUE TITLE
Handling CountQueryOption

### ISSUE THREAD

**Issue by xaviervv** [AutoMapper/AutoMapper.Extensions.OData#4]
Found issue during investigation of #3 
$count mixed with $top will only return the $top rows, instead of all rows. 

Quick investigation: there is no count anywhere.
According to OData First execute the filter.
Then apply count before orderby/paging

```
res = options.Filter.ApplyTo()

if (options.Count?.Value == true)
{
    options.Request.ODataFeature().TotalCount = ((IQueryable<T>)res).LongCount();
}
```

**Comment by xaviervv** [comment]
@BlaiseD The above should fix it. But I have no idea where in the code to put this. 

Can't put it in the GetAsync or QueryAsync method with the overload options, as it would be too late in the flow. Results already fetched. 

Can't put it in the expression overload, I don't have access to options. 

I can only think of 2 things: add options as param in the expression overload, or make the expression overload return a tuple or new custom object, which contain the queryable / icollection + count. 

I would prefer the first solution, and make it optional, ... I think this would give the least conflicts with already existing code. 

I'll go ahead and implement it / create a pr. 

If you don't like it, you can change it; I will upload the project with various fixes to our own TFS server / nuget feed, so we can already use it.

**Comment by xaviervv** [comment]
@BlaiseD Broke the count with filtering on your myget.
I quickly created a few additional objects in order to test filtering with count. The count doesn't match, and always return the total amount of the query in the db.

If you look at my PR, I applied count after applying filtering, this was working.
You applied it together with the expand options.

One of my remarks in this PR was that I didn't really know where to put it in order to not change your  method signatures too much.

**Comment by BlaiseD** [comment]
Ok -  sorry I missed that.  I'll fix later or you can create a PR.  I propose getting the filter and calling LongCount separately via the [query method](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/6a0f98ae5a7bd690b9c68e73db7ae9f5c8d944e5/AutoMapper.AspNetCore.OData.EFCore/QueryableExtensions.cs#L162) or a separate dynamically generated LongCount expression.  Then it's independent of the rest of the query.

**Comment by BlaiseD** [comment]
Or just pass the filter to LongCount :)

**Comment by xaviervv** [comment]
I totally forgot that we can indeed use expressions in count -.-

I’ll see if I can do something tonight.

**Comment by BlaiseD** [comment]
@xaviervv  I was wrong on just passing the filter to `LongCount()` actually because `LongCount()` on the `IQueryable<TData>` instance takes a TData filter but the OData filter is TModel.

So you'll want to map the expression - but just doing that could look clunky.  I propose just doing something like the orderBy e.g. for `OrderBy` we're creating this `Expression<Func<IQueryable<TModel>, IQueryable<TModel>>> orderByLanbda = q => q.OrderBy(e => e.Field);`

For `LongCount()` the expression would be `Expression<Func<IQueryable<TModel>, long>> longLanbda = q => q.LongCount(e => e.Field > 100);` then I'd say pass the whole thing to the `QueryAsync` method which already maps the expression.
`public static async Task<TReturn> QueryAsync<TModel, TData, long, long, long>();`

Open to other options - if you have time I mean.

**Comment by BlaiseD** [comment]
Fixed with release 1.0.1-preview02.
